cms/djangoapps/contentstore/views/course.py

```diff
@@ -469,6 +472,60 @@ def _get_rerun_link_for_item(course_key):
     return reverse_course_url('course_rerun_handler', course_key)
 
 
+def _ora1_deprecation_info(course_id, advanced_modules):
+    """
+    Returns information about ORA1
+        * Modules presence in the Advanced Module List.
+        * Components presence in the course outline.
+
+    Arguments:
+        course_id (CourseKey): course id
+        advanced_modules (list): advance module list
+
+    Returns:
+        Dict with following keys:
+
+        ora1_enabled (bool): True if "peergrading" and/or "combinedopenended" are in the Advanced Module List else False
+        ora1_components (list): List of ora1 components and their parent's url
+        advance_settings_url (str): URL to advance settings page
+    """
+    default = {
+        'ora1_enabled': False,
+        'ora1_components': [],
+        'advance_settings_url': ''
+    }
+
+    modified_timestamp = CourseStructure.objects.filter(course_id=course_id).values('modified')
```

Comment:
I think the point of caching is to avoid hitting the database. How about we clear the cache on a `course_publish` event instead? That way we do not have to worry about the modification date.